### ISSUE TITLE
compute stats for ggplot2 after-stats functionality

### ISSUE THREAD

**Issue by corybrunson** [corybrunson/ggalluvial#50]
### Description of the issue

The ggplot2 stat layers `stat_count()` and `stat_sum()` add "count", "n", and/or "prop" columns to the data returned by their `compute_*()` functions, which enable the `after_stat()` functionality described [here](https://ggplot2.tidyverse.org/reference/aes_eval.html). While the ggalluvial layers, in particular `*_stratum()`, mimic much of the behavior of `geom_bar()`, the user is not able to control aesthetic evaluation in the same way.

### Reproducible example (preferably using `reprex::reprex()`)

[This question on StackOverflow](https://stackoverflow.com/q/60725388/4556798) illustrates this limitation. The failed solution attempt below should work once the functionality is implemented:

``` r
library(ggplot2)
library(ggalluvial)
# toy data set
df <- data.frame('id' = rep(1:50,2),
                 'stage' = c(rep(1,50), rep(2,50)),
                 'group' = sample(c('A','B','C'), 100, replace = TRUE))
# without labeling
ggplot(df,
       aes(x = stage, stratum = group, alluvium = id, fill = group)) +
  scale_x_discrete(expand = c(.1, .1)) +
  geom_flow() +
  geom_stratum(alpha = .5)
```

![](https://i.imgur.com/VfZSY0h.png)

``` r
# attempted labeling using `after_stat()`
ggplot(df,
       aes(x = stage, stratum = group, alluvium = id, fill = group)) +
  scale_x_discrete(expand = c(.1, .1)) +
  geom_flow() +
  geom_stratum(alpha = .5) +
  geom_text(aes(label = scales::percent(after_stat(prop))), stat = "stratum")
#> Error in after_stat(prop): object 'prop' not found
```

![](https://i.imgur.com/WKG2lz4.png)

<sup>Created on 2020-03-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by corybrunson** [comment]
This is underway on the new `compute-stats` branch.

**Comment by corybrunson** [comment]
Once implemented, this might obviate the need for the ad hoc class-dependent aggregation steps currently performed by the `stat_*()` layers.

**Comment by corybrunson** [comment]
The example below relies on the ad hoc aggregation of numeric variables. It cannot be reproduced using `after_stat()` because the variable `a` is mapped to the aesthetic `label`, which is not involved in the `stat_*()` calculations.

``` r
library(ggalluvial)
#> Loading required package: ggplot2
data(vaccinations)
levels(vaccinations$response) <- rev(levels(vaccinations$response))
ggplot(vaccinations,
       aes(x = survey, stratum = response, alluvium = subject,
           y = freq, fill = response, label = round(a, 3))) +
  geom_lode() + geom_flow() +
  geom_stratum(alpha = 0) +
  geom_text(stat = "stratum")
```

![](https://i.imgur.com/VkH7ucV.png)

<sup>Created on 2020-03-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

A solution would be to add the optional aesthetic `weight` to the stat layers, with default `weight = 1L`, which would weight `n` as in `stat_sum()` as well as `count` and `prop` as in `stat_count()`.

This would also resolve the theoretically serious problem that the values of `a` are rounded _before_ being summed.

**Comment by corybrunson** [comment]
As of 18f4091b0e577398ca51424c0b47fef91898d527, the above problems seem to be solved.

### Expressing computed variables in percentages

``` r
library(ggalluvial)
#> Loading required package: ggplot2
set.seed(1)
df <- data.frame('id' = rep(1:50,2),
                 'stage' = c(rep(1,50), rep(2,50)),
                 'group' = sample(c('A','B','C'), 100, replace = TRUE))
ggplot(df,
       aes(x = stage, stratum = group, alluvium = id, fill = group)) +
  scale_x_discrete(expand = c(.1, .1)) +
  geom_flow() +
  geom_stratum(alpha = .5) +
  geom_text(aes(label = scales::percent(after_stat(prop))), stat = "stratum")
```

![](https://i.imgur.com/wU2OI3c.png)

<sup>Created on 2020-03-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### Weighting computed variables

``` r
library(ggalluvial)
#> Loading required package: ggplot2
# rightward flow aesthetics for vaccine survey data
data(vaccinations)
levels(vaccinations$response) <- rev(levels(vaccinations$response))
# annotate with proportional counts
ggplot(vaccinations,
       aes(x = survey, stratum = response, alluvium = subject,
           y = freq, fill = response)) +
  geom_lode() + geom_flow() +
  geom_stratum(alpha = 0) +
  geom_text(stat = "stratum", aes(label = round(after_stat(prop), 3)))
```

![](https://i.imgur.com/BwF5cti.png)

``` r
# annotate with survey-weighted proportional counts
ggplot(vaccinations,
       aes(x = survey, stratum = response, alluvium = subject,
           y = freq, fill = response, weight = a)) +
  geom_lode() + geom_flow() +
  geom_stratum(alpha = 0) +
  geom_text(stat = "stratum", aes(label = round(after_stat(prop), 3)))
```

![](https://i.imgur.com/Vyhsrrr.png)

<sup>Created on 2020-03-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by corybrunson** [comment]
These were included in v0.12.0, which was released from 5ea1f2df20711416621521da4d5f815a998e717d.
